samples/AudioPlayerSample/ViewModels/AudioRecorderPageViewModel.cs

```diff
@@ -4,36 +4,117 @@ namespace AudioPlayerSample.ViewModels;
 
 public class AudioRecorderPageViewModel : BaseViewModel
 {
-	readonly IAudioManager audioManager;
-	IAudioRecorder? audioRecorder;
+   ContentPage page;
 
-	public AudioRecorderPageViewModel(
-		IAudioManager audioManager)
-	{
-		StartCommand = new Command(Start);
+   readonly IAudioManager audioManager;
+   IAudioRecorder audioRecorder;
+   IAudioPlayer audioPlayer;
+   IAudioSource audioSource = null;
 
-		this.audioManager = audioManager;
-	}
+   string recordButtonText = "Record";
+   public string RecordButtonText
+   {
+      get => recordButtonText;
+      set
+      {
+         recordButtonText = value;
+         NotifyPropertyChanged();
+      }
+   }
 
-	public Command StartCommand { get; set; }
+   Color recordButtonColor = Colors.Blue;
+   public Color RecordButtonColor
+   {
+      get => recordButtonColor;
+      set
+      {
+         recordButtonColor = value;
+         NotifyPropertyChanged();
+      }
+   }
 
-	async void Start()
-	{
-		audioRecorder = audioManager.CreateRecorder();
+   double audioTime = 0;
+   public double AudioTime
+   {
+      get => audioTime;
+      set
+      {
+         audioTime = value;
+         NotifyPropertyChanged();
+      }
+   }
 
-		try
-		{
-			await audioRecorder.StartAsync();
+   public AudioRecorderPageViewModel(IAudioManager audioManager)
+   {
+      StartCommand = new Command(Start);
+      PlayCommand = new Command(playAudio);
+      this.audioManager = audioManager;
+   }
 
-			await Task.Delay(5000);
+   public void setPage(ContentPage page)
+   {
+      try
+      {
+         this.page = page;
+      }
+      catch (Exception ex)
+      {
+         page.DisplayAlert("Alert", $"setPage Exception: {ex.Message}", "OK");
+      }
+   }
```

Comment:
The view model really should never know about the view. If there isn't a fancy interface provided by .NET MAUI in order to call `DisplayAlert` from a view model perhaps we should roll a little `IAlertService` or something